code/modules/mob/holder.dm

```diff
@@ -29,52 +25,28 @@
 
 /obj/item/weapon/holder/process()
 
-	if (!(istype(loc,/obj/item/weapon/storage)))//Mobs bug out if placed directly into a container
-		if(!get_holding_mob())
+	if(!get_holding_mob() || contained.loc != src)
+
+		for(var/mob/M in contents)
 
-			for(var/mob/M in contents)
+			var/atom/movable/mob_container
+			mob_container = M
+			mob_container.loc = src.loc//if the holder was placed into a disposal, this should place the animal in the disposal
```

Comment:
This into `mob_container.forceMove(src.loc)`

Also, why not just keep rolling with `M`?